client/batch.go

```diff
@@ -308,6 +315,29 @@ func (b *Batch) Scan(s, e interface{}, maxRows int64) {
 	b.initResult(1, 0, nil)
 }
 
+// ReverseScan retrieves the rows in reserve order starting at end (exclusive)
+// and finishing at begin(inclusive).
+//
+// A new result will be appended to the batch which will contain up to maxRows
+// rows and Result.Err will indicate success or failure.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (b *Batch) ReverseScan(s, e interface{}, maxRows int64) {
```

Comment:
For user/client , Scanning `("a", "b"]` (left-exclude, right-include) 
In our internally implementation , Scanning `["a".next(), "b".next())`  (left-include, right-exclude);
The conversion place is before `Send()` in kv/db.go .